mapper/cumulocity/c8y_translator_lib/src/serializer.rs

```diff
@@ -126,3 +126,171 @@ impl GroupedMeasurementVisitor for C8yJsonSerializer {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use assert_json_diff::*;
+    use serde_json::json;
+
+    use super::*;
+    use chrono::{offset::FixedOffset, DateTime, Local};
+    fn test_timestamp() -> DateTime<FixedOffset> {
+        let local_time_now: DateTime<Local> = Local::now();
+        local_time_now.with_timezone(local_time_now.offset())
+    }
+    #[test]
+    fn serialize_single_value_message() {
+        let mut serializer = C8yJsonSerializer::new().unwrap();
+        let timestamp = test_timestamp();
+        serializer.timestamp(timestamp).unwrap();
+        serializer.measurement("temperature", 25.5).unwrap();
+        let output = serializer.bytes().unwrap();
+
+        let expected_output = json!({
+            "type": "ThinEdgeMeasurement",
+            "time": timestamp.to_rfc3339(),
+            "temperature":{
+                "temperature":{
+                    "value": 25.5
+                }
+            }
+        });
+
+        assert_json_eq!(
+            serde_json::from_slice::<serde_json::Value>(&output).unwrap(),
+            expected_output
+        );
+    }
+    #[test]
+    fn serialize_multi_value_message() {
+        let mut serializer = C8yJsonSerializer::new().unwrap();
```

Comment:
try to get rid of these `unwrap`s (applies to all test cases).

One way would be to just use `anyhow::Result`:

```rust
#[test]
fn serialize_multi_value_message() -> anyhow::Result<()> {
   let mut serializer = C8yJsonSerializer::new()?;
   ...
   Ok(())
}
```